proto/destination.proto

```diff
@@ -166,9 +166,20 @@ message DestinationProfile {
 
   // If this list is non-empty, requests to this destination should instead be
   // split between the destinations in this list.  Each destination should
-  // receive a portion of the requests proportional to its weight.  If this list
-  // is empty, requests should be sent to this destination as normal.
+  // receive a portion of the requests proportional to its weight.  If this
+  // list is empty, requests should be sent to this destination as normal.
   repeated WeightedDst dst_overrides = 3;
+
+  // If this field is set, it indicates that the target is a single (pod)
+  // endpoint and not a service address. The values of `fully_qualified_name`
+  // and `dst_overrides` will be ignored for the purposes of service discovery--
+  // traffic split and load balancing will be skipped and the single endpoint
+  // are used.
+  //
+  // Similarly, if this is unset and both `fully_qualified_name` &
+  // `dst_overrides` are not set, then discovery and load balancing will be
+  // skipped.
+  WeightedAddr endpoint = 6;
```

Comment:
Is there any difference between returning no `fully_qualified_name` and no `endpoint` versus returning an `endpoint` that contains the requested address with no pod metadata (i.e. no identity, labels, etc.)?  In the case of a lookup that does not correspond to a pod or service, which of these two should the server return?

Reply:
On one hand, returning the endpoint is a more explicit indication from the control plane to forward the connection/request; but, on the other hand, the control plane really doesn't know anything about the endpoint.

I'd suggest the following:

* If the target refers to a known endpoint (i.e. the address of a pod, a node, or a value in an entry in an EndpointSlice), an explicit endpoint _should_ be returned.
* If the target does not refer to a known endpoint, an empty, default profile should be returned.

They're ]functionally equivalent](https://github.com/linkerd/linkerd2-proxy/pull/736/files#diff-e1387ffcc93e5f1847ef76987315bf8ad96e245fc49623b3117868761e79d2efR120-R127) today, so I think it's not a hard blocker if an endpoint isn't returned in all of these cases today.